sklearn/utils/validation.py

```diff
@@ -499,14 +499,38 @@ def check_array(array, accept_sparse=False, dtype="numeric", order=None,
         msg = ("Data with input dtype %s was converted to %s%s."
                % (dtype_orig, array.dtype, context))
         warnings.warn(msg, DataConversionWarning)
+
+    ## Indices Datatype regulation
+    sparse_indices_check(array,accept_large_sparse)
+
     return array
 
+def sparse_indices_check(array,accept_large_sparse=False):
+    """
+    Indices Regulation for CSR Matrices.
+    Only Int32 are supported for now
+
+    """
+    if accept_large_sparse==False:
+        if type(array) ==sp.csr.csr_matrix:
```

Comment:
Use `array.format`.

We also need to check for:
* `csc_matrix.indices`;
* `coo_matrix.row`
* `dia_matrix.offsets`

And for completeness's sake:
* `bsr_matrix.indices`

Reply:
will do for bsr_matrix directly